app/controllers/YourContactDetailsController.scala

```diff
@@ -42,6 +44,29 @@ class YourContactDetailsController @Inject()(authenticate: IdentifierAction,
   val log: LoggerLike = Logger(this.getClass)
 
   def onPageLoad(): Action[AnyContent] = authenticate async { implicit request =>
+
+    val localSessionId: SessionId = getSessionId()
+
+    localSessionId match {
+      case a => sessionCacheConnector.getSessionId(localSessionId.value) match {
```

Comment:
We need to make sure that when there is no session id in the headerCarriers it must not call sessionCacheConnector and just redirect to home page. Is there any other reason why we are calling sessionCacheConnector in that case?

Reply:
This will be taken care as part of refactoring task in next sprint